### ISSUE TITLE
Error: listen EPERM: operation not permitted 0.0.0.0:8085

### ISSUE THREAD

**Issue by Alfiegerner** [JanLochi/zigbee2mqtt-snap#3]
Hi,

Thought it better to do separate issues to help anyone else bumping into same problems, apologies for spam.

I'm now getting the following doing `sudo snap run`. Looks like permissions on listening to port 8085.

```
Zigbee2MQTT:info 2021-04-20 21:53:03: Started frontend on port 0.0.0.0:8085
events.js:291
      throw er; // Unhandled 'error' event
      ^

Error: listen EPERM: operation not permitted 0.0.0.0:8085
    at Server.setupListenHandle [as _listen2] (net.js:1316:16)
    at listenInCluster (net.js:1364:12)
    at doListen (net.js:1501:7)
    at processTicksAndRejections (internal/process/task_queues.js:85:21)
Emitted 'error' event on Server instance at:
    at emitErrorNT (net.js:1343:8)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  code: 'EPERM',
  errno: 'EPERM',
  syscall: 'listen',
  address: '0.0.0.0',
  port: 8085
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.18.1 start: `node index.js`
npm ERR! Exit status 1
```

I did `snap connect janlochi-zigbee2mqtt:network` and have the following snap connections:

```
sudo snap connections janlochi-zigbee2mqtt
Interface    Plug                              Slot        Notes
network      janlochi-zigbee2mqtt:network      :network    -
serial-port  janlochi-zigbee2mqtt:serial-port  :usbserial  manua
```

Any idea what I'm missing?

Thanks.

**Comment by JanLochi** [comment]
Hi @Alfiegerner ,

Thank you for taking the time to report your issue. I can confirm that there is an issue regarding the new frontend feature. I did not came around to test it, so I missed it.

Most likely I have to add the capability to open sockets to the snap. I will do so and report back to you.

**Comment by Alfiegerner** [comment]
Great thanks @JanLochi, I saw the zwavejs2mqtt has a network-bind plug which may do this.

Not related but if you don't mind another q - I was wondering about the build process and whether I could rely on this snap for updates (main reason for moving from native node compile).  Is there an automated CI type build that will continue updating versions?  I guess I'm asking if I need to keep an eye on versions and check if it is up to date.  Thanks a mil.

**Comment by JanLochi** [comment]
Yes, the network-bind plug was missing. Now the frontend is working on my side. Fixed with 2c0a954a20acd3de033e496fce14325890dda863
The build of the snap is available in the candidate as version `1.18.2-9`.
I will close this issue on your confirmation. Thanks :+1: 

> Not related but if you don't mind another q - I was wondering about the build process

I don't mind :smiley:. The build is automated with the snapcraft CI system. Each commit to this repo will trigger a build, which will be automatically placed in the edge channel on the snapstore.
After some (minimal) testing on my side, and not receiving any issues, I will promote the build to the stable channel. Depending on the channel the users are following, snapd will automatically refresh to the newer version. This happens four times a day. https://snapcraft.io/docs/keeping-snaps-up-to-date

However, new upstream releases of zigbee2mqtt do not trigger an update of the snap. It's a manual process, because upstream could break the snap. As it just did with de frontend feature.
Usually I will update to the newest upstream release within a week.
Feel free to open an issue, if you need a new version urgently.

**Comment by Alfiegerner** [comment]
Thank you @JanLochi , update works a treat 👍
